generators/app/prompts.js

```diff
@@ -62,16 +72,53 @@ async function askForApplicationType() {
     },
   ];
 
-  const answers = await this.prompt([
-    {
-      type: 'list',
-      name: 'applicationType',
-      message: `Which ${chalk.yellow('*type*')} of application would you like to create?`,
-      choices: applicationTypeChoices,
-      default: appDefaultConfig.applicationType,
-    },
-  ]);
-  this.applicationType = this.jhipsterConfig.applicationType = answers.applicationType;
+  await this.prompt(
+    [
+      {
+        type: 'list',
+        name: 'applicationType',
+        message: `Which ${chalk.yellow('*type*')} of application would you like to create?`,
+        choices: applicationTypeChoices,
+        default: appDefaultConfig.applicationType,
+      },
+      {
+        when: answers => {
+          const { applicationType } = answers;
+          const askForMicrofrontend = [GATEWAY, MICROSERVICE].includes(applicationType);
+          if (!askForMicrofrontend) {
+            answers.microfrontend = false;
+          }
+          return askForMicrofrontend;
+        },
+        type: 'confirm',
+        name: 'microfrontend',
+        message: `Do you want to enable ${chalk.yellow('microfrontends')}?`,
```

Comment:
Since *type* has asterisks around it, it seems like this option should too. 
```suggestion
        message: `Do you want to enable ${chalk.yellow('*microfrontends*')}?`,
```
<img width="1330" alt="Screen Shot 2022-06-07 at 22 23 50" src="https://user-images.githubusercontent.com/17892/172531422-3434b7c1-500c-4cad-bd3a-152921d52d0c.png">